azure-iot-device/azure/iot/device/common/models/proxy_options.py

```diff
@@ -26,9 +27,11 @@ def __init__(
          If it is not provided, authentication will not be used (servers may accept unauthenticated requests).
         :param proxy_password: (optional) This parameter is valid only for SOCKS5 servers and specifies the respective password for the username provided.
         """
+        if proxy_type not in [socks.HTTP, socks.SOCKS4, socks.SOCKS5]:
```

Comment:
I think we should probably not require it to be the `socks` library constants (which translate to 1, 2, and 3) but rather our own defined constants. As is, this forces the customer to import the `socks` library into THEIR code which is very unnecessary.

Furthermore, use of these specific constants is only relevant in the MQTT case, and this is supposed to be implementation agnostic.

Naturally we would have to still support the `socks` constants for backwards compatibility.

Thoughts?

Reply:
I'm not especially concerned about this definition leaking up.  Adding a different enumeration for the sake of architectural purity just means that the user has another set of possible values to worry about, and they have to distinguish between `socks.SOCKS4` and, say, `azure.iot.device.common.proxy_types.SOCKS4`, and they both mostly mean the same thing, but one is accepted in one place and the other is accepted in a different place, 

Some folks will be just fine with this, some folks will use them interchangeably and incorrectly, some folks will spend time being confused about the difference, and some folks will define their own enumeration (say `customer.sock_type`) and their own mapping functions (`customer_sock_type_to_pysocks_sock_type()` and `customer_sock_type_to_azure_sock_type()`)

Reply:
I'm more concerned about the fact that we take a dependency on `PySocks` just for a few `isinstance` checks. We don't actually call any code inside `PySocks`, but we make all of our users install it.

In my perfect world, we would remove `PySocks` from `setup.py` and rewrite this to be something like 'if "SOCKS4' in str(proxytype) or "SOCKS5" in str(proxytype) or "HTTP" in str(proxytype):`.

FWIW, it's only 78 KB, which is only 2% of our total footprint (3.8 MB), but it's still a dependency that we don't need.

Reply:
Sounds good. I'll see what I can do.